### PR TITLE
Separate sort and add custom sort org setting

### PR DESCRIPTION
**Pull request by paulfalgout**
Shortcut Story ID: [sc-32648]

- [x] Add Tests

### CODE REVIEW

File: src/js/entities-service/actions.js [hunk 1]
```diff
@@ -18,8 +18,8 @@ const Entity = BaseEntity.extend({
     ].join();
     return this.fetchModel(id, { data: { include } });
   },
-  fetchActions({ filter }) {
-    const data = { filter };
+  fetchActions({ filter, include }) {
+    const data = { filter, include };
```
Comment (nmajor25) [inline src/js/entities-service/actions.js:22]: What is the purpose of the `include` here? Asking out of curiosity.

Comment (shadowhand) [reply src/js/entities-service/actions.js:22]: Adding `include` is required for (eg) `?include=patient.patient-fields.scoring` to get the necessary data (value `ranking`) for Risk Sort.

Comment (shadowhand) [reply src/js/entities-service/actions.js:22]: https://roundingwell.slack.com/archives/C02U318AW/p1670864213694099

Comment (paulfalgout) [reply src/js/entities-service/actions.js:22]: It was a deprecated API on the endpoint, but it was added back so we can pass `patient.patient-field.foo`

Comment (nmajor25) [reply src/js/entities-service/actions.js:22]: Gotcha, thanks @paulfalgout @shadowhand

**Review by nmajor25 — APPROVED** [review]

File: src/js/apps/patients/worklist/worklist_sort.js [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    const fieldName = this.get('field_name');
+    const keys = this.get('field_key').split('.');
+    const direction = this.get('direction');
+
+    return (a, b) => {
+      const aValue = getEntityFieldValue(a.model, fieldName, keys);
+      const bValue = getEntityFieldValue(b.model, fieldName, keys);
+
+      return alphaSort(direction, aValue, bValue);
+    };
```
Comment (shadowhand) [inline src/js/apps/patients/worklist/worklist_sort.js:34]: This needs to be updated to check for a sort option called `type` (or some such) with a default of `"alpha"` that also allows for `"numeric"`.

See [this bug][1].

[1]: https://app.shortcut.com/roundingwell/story/32716/custom-sort-on-a-number-is-using-alpha-sort-rather-than-numeric

File: src/js/apps/patients/worklist/worklist_sort.js [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+    const fieldName = this.get('field_name');
+    const keys = this.get('field_key').split('.');
+    const direction = this.get('direction');
+
+    return (a, b) => {
+      const aValue = getEntityFieldValue(a.model, fieldName, keys);
+      const bValue = getEntityFieldValue(b.model, fieldName, keys);
+
+      return alphaSort(direction, aValue, bValue);
```
Comment (shadowhand) [inline src/js/apps/patients/worklist/worklist_sort.js:33]: To resolve the issue with #879, this should read:

```js
return alphaSort(direction, aValue ?? '', bValue ?? '');
```

But to support `type: numeric` that would expand to something like:

```js
if (type === 'numeric') {
  return numSort(direction, aValue ?? 0, bValue ?? 0);
}

return alphaSort(direction, aValue ?? '', bValue ?? '');
```

File: src/js/apps/patients/worklist/worklist_sort.js [hunk 4]
```diff
@@ -0,0 +1,140 @@
+import { get, union, map, uniq, compact } from 'underscore';
+import Backbone from 'backbone';
+import Radio from 'backbone.radio';
+
+import { alphaSort } from 'js/utils/sorting';
```
Comment (shadowhand) [inline src/js/apps/patients/worklist/worklist_sort.js:5]: Need `numSort` here.

**Pull request merged**